generators/pkg/generator/generate.go

```diff
@@ -23,12 +23,17 @@ type GitHub struct {
 
 // GithubSpec defines Github spec
 type GitHubSpec struct {
-	URL   string         `json:"url,omitempty"`
-	Steps []v1beta1.Step `json:"steps,omitempty"`
+	URL                string         `json:"url,omitempty"`
+	Branch             string         `json:"branch,omitempty"`
+	Storage            string         `json:"storage,omitempty"`
+	SecretName         string         `json:"secretName,omitempty"`
+	SecretKey          string         `json:"secretKey,omitempty"`
+	ServiceAccountName string         `json:"serviceAccountName,omitempty"`
+	Steps              []v1beta1.Step `json:"steps,omitempty"`
```

Comment:
It'd be great to document these fields in the read me - what are the expected values, what are there default values, etc.